airgun/views/host_new.py

```diff
@@ -254,12 +254,117 @@ class traces(Tab):
         enable_traces = OUIAButton('OUIA-Generated-Button-primary-1')
 
     @View.nested
-    class ansible(Tab):
+    class insights(Tab):
         pass
 
     @View.nested
-    class insights(Tab):
-        pass
+    class ansible(Tab):
+        """View comprising the subtabs under the Ansible Tab"""
+
+        ROOT = './/div'
+
+        @View.nested
+        class roles(Tab):
+            TAB_NAME = 'Roles'
+            ROOT = './/div[@class="ansible-host-detail"]'
+
+            assignedRoles = Text('.//a[contains(@href, "roles/all")]')
+            edit = Button(locator='.//button[@aria-label="edit ansible roles"]')
+            table = Table(
+                locator='.//table[contains(@class, "pf-c-table")]',
+                column_widgets={'Name': Text('.//a')},
+            )
+            pagination = Pagination()
+
+        @View.nested
+        class variables(Tab):
+            TAB_NAME = 'Variables'
+            ROOT = './/div[@class="ansible-host-detail"]'
+            table = Table(
+                locator='.//table[contains(@class, "pf-c-table")]',
+                column_widgets={
+                    'Name': Text('.//a'),
+                    'Ansible role': Text('./span'),
+                    'Type': Text('./span'),
+                    # the next field can also be a form group
+                    'Value': Text('./span'),
+                    'Source attribute': Text('./span'),
+                    # The next 2 buttons are hidden by default, but appear in this order
+                    5: Button(locator='.//button[@aria-label="Cancel editing override button"]'),
+                    6: Button(locator='.//button[@aria-label="Submit override button"]'),
+                    # Clicking this button hides it, and displays the previous 2
+                    7: Button(locator='.//button[@aria-label="Edit override button"]'),
```

Comment:
Should we have dev make this easier for automation like a better id to select by instead of titles (since those can be changed)? We can address this in another PR if you think we need to. In terms of automation coverage I imagine you wouldn't really be using these buttons much anyway?

Reply:
It's always better for dev to have specific IDs for things, since they make it much more likely to not fall apart at small changes in the future. As far as automation coverage is concerned, no, you wouldn't.